src/irmin-pack/unix/stats.ml

```diff
@@ -171,13 +171,70 @@ module Index = struct
   let export m = Metrics.state m
 end
 
-type t = { pack_store : Pack_store.stat; index : Index.stat }
+module File_manager_stats = struct
+  type Metrics.origin += File_manager_stats
 
-let s = { pack_store = Pack_store.init (); index = Index.init () }
+  type t = {
+    mutable dict_flushes : int;
+    mutable suffix_flushes : int;
+    mutable index_flushes : int;
+    mutable auto_dict : int;
+    mutable auto_suffix : int;
+    mutable auto_index : int;
+    mutable flush : int;
+  }
+  [@@deriving irmin]
+
+  (* NOTE return a new instance each time, since fields are mutable *)
+  let create () =
+    {
+      dict_flushes = 0;
+      suffix_flushes = 0;
+      index_flushes = 0;
+      auto_dict = 0;
+      auto_suffix = 0;
+      auto_index = 0;
+      flush = 0;
+    }
+
+  (* type [stat] is an abstract type in stats.mli; FIXME what is it for? *)
+  type stat = t Metrics.t
+
+  let init () : stat =
+    Metrics.v ~origin:File_manager_stats ~name:"file_manager_metric"
+      ~initial_state:(create ()) t
+
+  (* [export] reveals the [t] contained in the [Metrics.t] container *)
+  let export : stat -> t = fun m -> Metrics.state m
```

Comment:
Why not just:
```ocaml
let export = Metrics.state m
```
?

Reply:
It's written eta-expanded elsewhere (admittedly with the m argument before the =). I don't think it really matters. I wanted to have the explicit type for export, so it was clear what the types were at that point... otherwise, it is a little hard to read.